src/univariate/continuous/normal.jl

```diff
@@ -1,12 +1,18 @@
-immutable Normal <: ContinuousUnivariateDistribution
-    μ::Float64
-    σ::Float64
+immutable Normal{T<:Number} <: ContinuousUnivariateDistribution
+    μ::T
+    σ::T
 
-    Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); new(μ, σ))
-    Normal(μ::Real) = Normal(μ, 1.0)
-    Normal() = Normal(0.0, 1.0)
+    Normal(μ::T, σ::T) = new(μ, σ)
 end
 
+Normal{T<:Number}(μ::T, σ::T) = Normal{T}(μ, σ)
+Normal(μ::Number, σ::Number) = Normal(promote(μ, σ)...)
+Normal(μ::Number) = Normal(μ, one(μ))
+#Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); Normal(μ, σ))
+Normal(μ::Real) = Normal(μ, one(μ))
+Normal() = Normal(0.0, 1.0)
+Normal(μ::Complex, σ::Complex) = error("Normal with complex mean and variance not defined")
```

Comment:
@StefanKarpinski, I a-lot agree with what you wrote.

I mentioned elsewhere but I'll repeat it here. If `Dual{T<:Real}<:Real`, then I think it's only consistent to have also `real(x::Dual{T<:Real})=x`. There will then have to be a different accessor for the non-epsilon component of the Dual number.

However, note that you could then have Dual{Dual{Float64}}, for example. I think this cannot be made to behave like a hyperdual numbers (with four parts: `noneps`, `eps1`, `eps2`, and `eps1eps2`), so it's probably a nonsense type.

However, also note that you could then have `Complex{Dual}` (which makes sense if you ask me, and I think would work today) and `Quaternion{Dual}` (which I haven't played with). I don't know what to do about the fact that `Dual{Complex}` (which would require a definition `Dual{T<:Number}`  instead of `Dual{T<:Real}`) should be essentially the same as `Complex{Dual}`.